packages/auth/src/core/auth/auth_impl.test.ts

```diff
@@ -786,4 +786,112 @@ describe('core/auth/auth_impl', () => {
       expect(auth._getRecaptchaConfig()).to.eql(cachedRecaptchaConfigOFF);
     });
   });
+
+  describe('AuthStateReady', () => {
+    let user: UserInternal;
+    let authStateChangedSpy: sinon.SinonSpy;
+
+    beforeEach(async () => {
+      user = testUser(auth, 'uid');
+
+      authStateChangedSpy = sinon.spy(auth, 'onAuthStateChanged');
+
+      await auth._updateCurrentUser(null);
+    });
+
+    it('immediately returns resolved promise if the user is previously logged in', async () => {
+      await auth._updateCurrentUser(user);
+
+      // authStateReady calls onAuthStateChanged, if there is no currentUser available
```

Comment:
Comments in this file are not publicly visible, iiuc.  Please correct if wrong, thanks!